packages/tools/src/eventDispatchers/shared/customCallbackHandler.ts

```diff
@@ -37,37 +38,43 @@ export default function customCallbackHandler(
     return false;
   }
 
-  // TODO: Filter tools by interaction type?
-  /**
-   * Iterate tool group tools until we find a tool that is:
-   * - active
-   * - has the custom callback function
-   *
-   */
-  let activeTool;
-  const toolGroupToolNames = Object.keys(toolGroup.toolOptions);
-
-  for (let j = 0; j < toolGroupToolNames.length; j++) {
-    const toolName = toolGroupToolNames[j];
-    const tool = toolGroup.toolOptions[toolName];
-    // TODO: Should be getter
-    const toolInstance = toolGroup.getToolInstance(toolName);
-
-    if (
-      // TODO: Should be enum?
-      tool.mode === Active &&
-      // TODO: Should be implements interface?
-      // Weird that we need concrete instance. Other options to filter / get callback?
-      typeof toolInstance[customFunction] === 'function'
-    ) {
-      activeTool = toolGroup.getToolInstance(toolName);
-      break;
-    }
-  }
+  const activeTool = getActiveTool(toolGroup, customFunction);
 
   if (!activeTool) {
     return;
   }
 
   activeTool[customFunction](evt);
 }
+
+// TODO: Filter tools by interaction type?
+/**
+ * Iterate tool group tools until we find a tool that is:
+ * - active
+ * - has the custom callback function
```

Comment:
How about:
- has an attribute with key 'customFunctionName' and whose value is a function.

I just was expecting customFunction to be a function itself, and it was weird that it wasn't.  Renaming it to customFunctionName and indicating how it works is useful.